utils/include/synchronized.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SYNCHRONIZED_H_INCLUDED
+#define SYNCHRONIZED_H_INCLUDED
+
+#pragma once
+
+#include <mutex>
+using namespace std;
+
+template <typename T>
+class Synchronized {
+public:
+  explicit Synchronized(T initial = T())
+    : value(move(initial))
+  {
+  }
+
+  struct Access {
+    T& ref_to_value;
+    lock_guard<mutex> guard;
+  };
+
+  Access GetAccess() {
+    return {value, lock_guard(m)};
+  }
+
+private:
+  T value;
```

Comment:
Maybe if the field was a reference type would that be a better option? Also, the constructor should be changed